scripts/stav-covid.py

```diff
@@ -0,0 +1,153 @@
+	parts = string.split(":")
+	if len(parts) == 1:
+		return datetime.datetime.combine(date, datetime.time(hour=int(parts[0])))
+	elif len(parts) == 2:
+		return datetime.datetime.combine(date, datetime.time(hour=int(parts[0]), minute=int(parts[1])))
+	else:
+		return None
+
+# Start with an empty schedule set
+schedules = {}
+
+# For each of the HTML tables we found,
+for table in tables:
+	print("Processing a table...")
+
+	# Find the prior sibling, which is an h3 containing the day in some format.
+	# Each table should have an h3 before it that has date information in it.
+	header = table.find_previous_sibling("h3").get_text().strip()
+	print('Found a header sibling with text "{}"'.format(header))
+
+	# Search the text for just the date
+	matching_text = date_re.match(header).group(0)
+
+	# NOTE(rye): Parses text like "Sunday, August 23"
+	date = datetime.datetime.strptime(matching_text, "%A, %B %d")
+
+	# Move the date to the current year
+	date = datetime.datetime(date.today().year, date.month, date.day)
+
+	# If the computed date is more than six months before today, assume it actually belongs in the next year.
+	if date < date.today() + datetime.timedelta(days=-180):
+		date = datetime.datetime(date.year + 1, date.month, date.day)
+
+	# We only know the date, so let's back out to that.
+	date = date.date()
+
+	print("Inferred date of {}; processing table now.".format(date))
+
+	body = table.find("tbody")
+
+	# NOTE(rye): Should only have one thead
+	meals = [meal_name_from(th) for th in table.find("thead").find("tr").find_all("th")]
+
+	for idx, meal in enumerate(meals):
+		if idx == 0:
+			continue
+		for row in body.find_all("tr"):
+			cols = [col.get_text() for col in row.find_all("td")]
+			if len(cols) != len(meals):
+				continue
+			if len("".join(cols)) == 0:
+				continue
+			dorm = cols[0]
+			# meal = meal
+			timespec = cols[idx]
+
+			print("Processing table entry: {} gets {} at {} on {}".format(dorm, meal, timespec, date))
+
+			# Split the timespec on the timespec_re matches
+			times = [time.strip() for time in timespec_re.split(timespec)]
+
+			if len(times) != 2:
+				warnings.warn("timespec {} did not split nicely".format(timespec))
+				continue
+
+			t_open = timeparse(times[0], date=date)
+			t_close = timeparse(times[1], date=date)
+
+			# Lunch always occurs after 6am.  If we have something else, we're
+			# probably 12 hours off.
+			if meal == "Lunch" and (t_open.hour < 6 or t_close.hour < 6):
+				t_open += datetime.timedelta(hours=12)
+				t_close += datetime.timedelta(hours=12)
+
+			# Dinner always occurs in the evening.  If we have something else,
+			# we're probably 12 hours off.
+			if meal == "Dinner" and (t_open.hour < 12 or t_close.hour < 12):
+				t_open += datetime.timedelta(hours=12)
+				t_close += datetime.timedelta(hours=12)
+
+			# The open time should not be the same or even close to the close
+			# time, so we should wrap it around.  (We do this by bumping up the )
```

Comment:
Bumping up the ____ (close time)?

```suggestion
			# time, so we should wrap it around.  (We do this by bumping up the 
			# close time)
```